Components/Coroutines/Town/TrinityTownRun.StashItems.cs

```diff
@@ -118,62 +118,58 @@ public static bool ShouldStash(TrinityItem i)
             return !i.IsProtected() && Combat.TrinityCombat.Loot.ShouldStash(i);
         }
 
-        public static async Task<bool> StashItems()
+        public static async Task<CoroutineResult> StashItems()
         {
             if (!ZetaDia.IsInTown)
-                return true;
-
-            var stashItems = Core.Inventory.Backpack.Where(ShouldStash).ToList();
-            if (!stashItems.Any())
-                return true;
+                return CoroutineResult.NoAction;
 
-            Core.Logger.Verbose($"[StashItems] Now to stash {stashItems.Count} items");
-            stashItems.ForEach(i => Core.Logger.Debug($"[StashItems] Stashing: {i.Name} ({i.ActorSnoId}) InternalName={i.InternalName} Ancient={i.IsAncient} Ann={i.AnnId}"));
+            var item = Core.Inventory.Backpack.FirstOrDefault(ShouldStash);
+            if (item == null)
+                return CoroutineResult.Done;
 
             if (!UIElements.StashWindow.IsVisible)
             {
                 var stash = ZetaDia.Actors.GetActorsOfType<GizmoPlayerSharedStash>().FirstOrDefault();
                 if (stash == null)
-                    return false;
+                    return CoroutineResult.Failed;
 
-                if (!await CommonCoroutines.MoveAndInteract(stash, () => UIElements.StashWindow.IsVisible))
-                    return false;
+                if (await CommonCoroutines.MoveAndInteract(
+                        stash,
+                        () => UIElements.StashWindow.IsVisible) == CoroutineResult.Running)
+                {
+                    return CoroutineResult.Running;
+                }
             }
 
             if (Core.Settings.Items.BuyStashTabs && StashPagesAvailableToPurchase)
             {
-                Core.Logger.Error("[StashItems] Attempting to buy stash pages");
+                s_logger.Error($"[{nameof(StashItems)}] Attempting to buy stash pages");
                 InventoryManager.BuySharedStashSlots();
             }
 
             // TODO: Figure out if those 2 calls are still relevant.
             await StackRamaladnisGift();
             await StackCraftingMaterials();
 
-            // Get the first item to stash...
-            var item = Core.Inventory.Backpack.FirstOrDefault(ShouldStash);
-            if (item == null)
-                return true;
-
             try
```

Comment:
should do a while loop here for item `var item = Core.Inventory.Backpack.FirstOrDefault(ShouldStash)` and Yield after each. Whole bot reruns until this point after a single item gets stashed. This looks absurd to me.